components/follow-list/FollowList.tsx

```diff
@@ -0,0 +1,98 @@
+import { followInfoDataTypes } from "@/types/User";
+import Image from "next/image";
+import React from "react";
+import * as styles from "../../styles/FollowListPage/FollowList/FollowList.css";
+import { useSearchParams } from "next/navigation";
+import usePutFollower from "@/hooks/follower/usePutFollower";
+import useDeleteFollower from "@/hooks/follower/useDeleteFollower";
+
+const FollowList = ({ followData }: { followData: followInfoDataTypes }) => {
+  const searchParams = useSearchParams();
+  const userToken = searchParams.get("token");
+  const putMutation = userToken && usePutFollower();
+  const deleteMutation = userToken && useDeleteFollower();
+
+  const { followingData, followersData } = followData;
+
+  // 맞팔 중인 사람들 리스트
+  const matchedList = followersData.filter((follower) => {
+    return followingData.some(
+      (following) => following.login === follower.login
+    );
+  });
```

Comment:
followersData 의 길이가 아주 길 수 있는 상황에서, 매 렌더링마다 다시 실행되면 비효율적이기 때문에 useMemo 를 적용하면 좋을 것 같아요!

밑의 unfollowingList 도 마찬가지에요.

```suggestion
  const matchedList = useMemo(() => followersData.filter((follower) => {
    return followingData.some(
      (following) => following.login === follower.login
    );
  }), [followersData, followingData]);
```